ServantHDBC/src/Api/Product.hs

```diff
@@ -0,0 +1,94 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeOperators #-}
+
+
+module Api.Product (ProductAPI, productHandlers) where
+
+import Control.Monad.Trans.Except
+import Data.Aeson
+import GHC.Generics
+import Servant
+
+
+data Product = Product
+    { ids               :: String
+    , q                 :: String
+    , title             :: String
+    , sku               :: String
+    , prod_type         :: String
+    , tags              :: String
+    , created_at_min    :: String
+    , created_at_max    :: String
+    , updated_at_min    :: String
+    , updated_at_max    :: String
+    , limit             :: String
+    , limit_count       :: Int
+    , offset            :: Int
+    , orderby           :: String
+    , fields            :: String
+    } deriving (Generic)
+
+instance FromJSON Product 
+
+type ProductResult = String
+type ProductAPI = "products" :>
+                (Capture "product_id" Int :>
+                    QueryParam "fields" String :>
+                    QueryParam "photo_sizes" String :>
+                    QueryParam "varants.photo_sizes" ProductResult :>
+                        Get '[JSON] String 
+            :<|> QueryParam "ids" String :>
```

Comment:
I added simple datatypes. It's possible to do much more sophisticated parsing. This is simply a string.